### DISCUSSION TITLE
How do I export a rescaled mesh+skeleton without it getting deformed?

### DISCUSSION THREAD

**Discussion by azufelt00** [BadDogSkyrim/PyNifly#218] [Q&A]
I'm pretty much a noob at Blender so forgive me if this is a dumb question.

Niftools let me specify a scale on import and export, so if I wanted something half-sized I imported it at 0.5 and exported at 1.0.

I don't see an option like that in PyNifly, so I imported a Skyrim nif as normal, resized it to 0.5 in blender, did my posing, then exported to Skyrim, but the result becomes stretched and deformed if I manipulated or posed the mesh in any way, even though everything looks normal in blender

Any idea what I'm doing wrong? Did I mess up scaling or exporting?

**Comment by BadDogSkyrim** [comment]
Yeah, I didn't do a scale on export, thinking you could just scale in Blender.

Scaling skinned meshes in Blender is confusing because of how the mesh transform interacts with the armature transform. Best way to clean up is: unparent the mesh; remove the armature modifier; apply all transforms to both mesh and armature; reassign the armature modifier. 

That should export cleanly.

**Reply by azufelt00** [reply]
Thank you for the response!! I need to test it but it sounds like it should work!

**Comment by Ivanvillad** [comment]
This a late answer, I know.
You are in trouble by exporting" as is" a previously downscaled model since the animations use absolute translation on various bones.
Therefore, when calculating the deformed animated vertex groups attached to those bones, the translation is applied on vertices in absolute (not considering any scaling made), resulting in a visual stretch.

You have no way to do what you want (your problem is centered on the skeleton), unless:
- you export as original scale (1.0) and scale down your model directly in the game, depending on what you are doing it may work nicely; think about it: pynifly does the same when importing in blender scale, this scale is globally applied (then removed when exporting), this very mechanism is applied in the game when you specify the height of a race or the scale of an actor
- you recreate *all* animations used by your model, applying your scale factor in the graph editor on all bones subject to translation (having Location FCurves) > Key > Transform > Scale > Y > enter your scale factor > check result (needless to say, you have then to map your new animations in CK for your model) => you are creating a new race/creature

Cleaning up all the human game animations by removing translations neither is a good (crazy but still valid) way, since the center of mass is always adjusted to give a realistic look (mostly the sway). Alas, this sway is given in absolute translation. Without, you'd transform your model in a robot and all positioning by offset (on furniture for example) will be wiped out.
